src/ghci/stderr.rs

```diff
@@ -87,11 +97,18 @@ impl GhciStderr {
     }
 
     #[instrument(skip(self), level = "trace")]
-    async fn ingest_line(&mut self, line: String) {
+    async fn ingest_line(&mut self, mut line: String) -> miette::Result<()>
+    where
+        W: AsyncWrite + Unpin,
+    {
         tracing::debug!(line, "Read stderr line");
+        line.push('\n');
         self.buffer.push_str(&line);
-        self.buffer.push('\n');
-        eprintln!("{line}");
+        self.writer
+            .write_all(line.as_bytes())
+            .await
+            .into_diagnostic()?;
+        Ok(())
     }
```

Comment:
I removed `self.buffer.push('\n');` in favor of `line.push('\n');`. `write_all` wouldn't add a trailing newline like `eprintln` did, so I need the newline in both places.